core/trino-main/src/main/java/io/trino/server/security/HeaderBasedAuthenticatorConfig.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.validation.FileExists;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public class HeaderBasedAuthenticatorConfig
+{
+    private static final Splitter SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+
+    private boolean treatHeadersCaseSensitive;
+    private List<String> headersToExtract = ImmutableList.of("x-forwarded-client-cert");
+    private Optional<String> userMappingPattern = Optional.empty();
+    private Optional<File> userMappingFile = Optional.empty();
+    private List<File> headerBasedAuthenticatorFiles = ImmutableList.of(new File("etc/header-based-authenticator.properties"));
+
+    @Config("http-server.authentication.header-based.treat-headers-as-case-sensitive")
+    @ConfigDescription("Whether or not to treat the request context headers as case sensitive during extraction")
+    public HeaderBasedAuthenticatorConfig setTreatHeadersCaseSensitive(boolean treatHeadersCaseSensitive)
+    {
+        this.treatHeadersCaseSensitive = treatHeadersCaseSensitive;
+
+        return this;
+    }
+
+    public boolean getTreatHeadersCaseSensitive()
+    {
+        return this.treatHeadersCaseSensitive;
+    }
+
+    @Config("http-server.authentication.header-based.headers-to-extract")
+    @ConfigDescription("List of headers to be extracted from the request context and passed to the authenticator")
+    public HeaderBasedAuthenticatorConfig setHeadersToExtract(String headersToExtract)
```

Comment:
Instead of using splitter I think airlift will also accept `List<String>` as parameter

Reply:
thanks, I will do the same for the configuration files in that case